src/app/features/course-edit/edit-course.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import {CoursesService} from "../../services/courses.service";
+import {Course} from "../models";
+import {ActivatedRoute, Router} from "@angular/router";
+
+@Component({
+  // selector: 'course-edit',
+  templateUrl: './new-course.component.html',
+  styleUrls: ['../course/course.component.css', '../../shared/components/shared.style.css']
+})
+export class EditCourseComponent implements OnInit {
+
+  btnTextCreate = 'Create course';
+  btnTextEdit = 'Save changes';
+  btnAuthorText = 'Create author';
+  btnDelAuthorText = 'Delete author';
+  hasError = false;
+  hasError1 = false;
+  hasError2 = false;
+  hasError3 = false;
+  private course: Course;
+
+  form: FormGroup;
+
+  constructor(private coursesService: CoursesService, private route: ActivatedRoute, private router: Router) {
+    const id = Number.parseInt(this.route.snapshot.paramMap.get('id'));
+    console.log('id = ' + id);
+    this.course = coursesService.getCourse(id);
+    this.authors = this.course.authors
+    this.form = new FormGroup({
+      courseTitle: new FormControl(this.course.courseTitle),
+      author: new FormControl(null),
+      duration: new FormControl(this.course.duration),
+      created: new FormControl(this.course.created),
+      text: new FormControl(this.course.text)
+    });
```

Comment:
use ngOnInit lifecycle hook instead of constructor, and read the difference of contstructor and ngOnInit